lib/internal/modules/esm/loader.js

```diff
@@ -391,6 +396,7 @@ class ModuleLoader {
       conditions: this.#defaultConditions,
       importAttributes,
       parentURL,
+      importMap: this.importMap,
```

Comment:
I don't think this is transferable, meaning this will throw as soon as any custom loader is registered. Can you add a test with a loader?

Reply:
Yeah, this currently doesn't work. Frankly I am not sure *how* we do want it to work. Do we want to pass just the "raw" import map object and add handling inside the worker to re-hydrate the `ImportMap` class with it? I am struggling to decide if there is a good use case to expose this to other loaders. Would it instead make sense to remove it from `context` before passing context to the worker?

Reply:
Yup. Policies have to xfer the source. Take note though that if Workers are allowed to override the value in your design (since they can manipulate argv)

Reply:
Can you represent the import map value as a plain object that’s transferable? Like I would assume you should be able to implement `toJSON` and `fromJSON` methods to your class, and JSON is transferable. Not that I’m suggesting going as far as JSON when an object will do, but if `toJSON`/`fromJSON` is possible then some kind of transferable data structure is therefore also possible.

Reply:
> Can you represent the import map value as a plain object that’s transferable?

Yep, that is what I meant to say by "Do we want to pass just the raw import map object and add handling inside the worker to re-hydrate the ImportMap class". I just am not sure this makes sense to do. Not only would it be slow but I also am not sure I can think of a use case for accessing/changing the import map in a loader.

Reply:
Ok, so the initial non-transferable thing was actually the parent URL instance. I resolved that. Now we have a new question of if, when using the customized loader, if the import map resolve should happen inside the worker or on the main thread.

I just pushed an implementation where it happens on the main thread, which means that the processing is on the main thread both when using a custom loader and not. What are the benefits of doing the import map resolution in the worker thread.

cc @JakobJingleheimer maybe we continue this conversation here instead of slack?

Reply:
Ok, in slack I think @GeoffreyBooth made a good case for this needing to be on `context`. I think that means I need to make it load the import map inside the worker when using custom loader hooks.

I still think I can avoid reading it twice by checking if there are customization and only loading it on the main thread when there are not.